spec/features/guide_spec.rb

```diff
@@ -95,16 +95,64 @@
     expect(edition.published?).to eq true
   end
 
-  it "shows api errors" do
-    fill_in_guide_form
+  context "when creating a new guide" do
+    context "with an api that fails straight away" do
+      before do
+        api_error = GdsApi::HTTPClientError.new(422, "Error message stub", "error" => { "message" => "Error message stub" })
+        expect_any_instance_of(GdsApi::PublishingApiV2).to receive(:put_content).and_raise(api_error)
+      end
+
+      it "shows api errors" do
+        fill_in_guide_form
+        click_button "Save Draft"
+
+        within ".alert" do
+          expect(page).to have_content('Error message stub')
+        end
+      end
+
+      it "does not store a guide" do
+        fill_in_guide_form
+        click_button "Save Draft"
+
+        expect(Guide.count).to eq 0
+        expect(Edition.count).to eq 0
+      end
+    end
+  end
 
-    api_error = GdsApi::HTTPClientError.new(422, "Error message stub", "error" => { "message" => "Error message stub" })
-    expect_any_instance_of(GdsApi::PublishingApiV2).to receive(:put_content).and_raise(api_error)
+  context "when updating a guide" do
+    context "with an api that fails the second time" do
+      let :api_error do
+        GdsApi::HTTPClientError.new(422, "Error message stub", "error" => { "message" => "Error message stub" })
+      end
 
-    click_button "Save Draft"
+      it "shows api errors" do
+        edition = Generators.valid_edition(title: "something")
+        guide = Guide.create!(slug: "/service-manual/something", latest_edition: edition)
 
-    within ".alert" do
-      expect(page).to have_content('Error message stub')
+        expect_any_instance_of(GuidePublisher).to receive(:process).once.and_raise(api_error)
+
+        visit edit_guide_path(guide)
+        click_button "Save Draft"
+
+        within ".alert" do
+          expect(page).to have_content('Error message stub')
+        end
+      end
+
+      it "does not store a guide" do
+        edition = Generators.valid_edition(title: "something")
+        guide = Guide.create!(slug: "/service-manual/something", latest_edition: edition)
+
+        expect_any_instance_of(GuidePublisher).to receive(:process).once.and_raise(api_error)
+
+        visit edit_guide_path(guide)
```

Comment:
should we try to change a field, try saving and then assert that the field did not change?